### DISCUSSION TITLE
RENOVATE_REGEX_MANAGERS env could not parsed

### DISCUSSION THREAD

**Discussion by Admirari80** [renovatebot/renovate#20673] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:34.153.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.6.2

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to outsource the regexManagers configuration to an environment variable that affects to all GitLab projects.

```json
  "regexManagers": [
    {
      "description": "Docker versions in puppet yaml files",
      "fileMatch": [
        "(.*)?data\\/.*y(a)?ml$"
      ],
      "matchStringsStrategy": "combination",
      "matchStrings": [
        "-*?_image:\\s*(?<depName>.*)\\s",
        "-*?_version:\\s*(?<currentValue>.*)\\s"
      ],
      "datasourceTemplate": "docker"
    },
    {
      "description": "Docker versions in puppet yaml files",
      "matchStrings": [
        "\\s*image:\\s*(?<depName>.*):(?<currentValue>.*)\\s"
      ],
      "fileMatch": [
        "(.*)?data\\/.*y(a)?ml$"
      ],
      "datasourceTemplate": "docker"
    },
    {
      "fileMatch": ["(^|/|\\.)Dockerfile$", "(^|/)Dockerfile\\.[^/]*$"],
      "matchStrings": [
        "# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s"
      ],
      "versioningTemplate": "{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"
    }
  ]
```

The documentation says: 

<h2 id="regexmanagers" style="box-sizing: inherit; font-weight: 300; letter-spacing: -0.01em; font-size: 1.5625em; line-height: 1.4; margin: 1.6em 0px 0.64em; --md-scroll-margin:3.6rem; --md-scroll-offset:0.2rem; scroll-margin-top: calc(var(--md-scroll-margin) - var(--md-scroll-offset)); color: rgba(0, 0, 0, 0.87); font-family: Roboto, -apple-system, BlinkMacSystemFont, Helvetica, Arial, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">regexManagers<a class="headerlink" href="https://docs.renovatebot.com/configuration-options/#regexmanagers" title="Permanent link" style="box-sizing: inherit; -webkit-tap-highlight-color: transparent; color: var(--md-accent-fg-color); text-decoration: none; word-break: break-word; transition: color 0.25s ease 0s, opacity 125ms ease 0s; display: inline-block; opacity: 1; margin-left: 0.5rem;">¶</a></h2><p style="box-sizing: inherit; color: rgba(0, 0, 0, 0.87); font-family: Roboto, -apple-system, BlinkMacSystemFont, Helvetica, Arial, sans-serif; font-size: 19.2px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Custom managers using regex matching.</p><div class="md-typeset__scrollwrap" style="box-sizing: inherit; margin: 1em -0.8rem; overflow-x: auto; touch-action: auto; color: rgba(0, 0, 0, 0.87); font-family: Roboto, -apple-system, BlinkMacSystemFont, Helvetica, Arial, sans-serif; font-size: 19.2px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="md-typeset__table" style="box-sizing: inherit; display: inline-block; margin-bottom: 0.5em; padding: 0px 0.8rem;">

Name | Value
-- | --
type | array
subType | object
cli | --regex-managers
mergeable | true
env | RENOVATE_REGEX_MANAGERS

### What you tried so far.

I tried to store the above code

```bash
jq  -n -c '{
      "fileMatch": [
        "(.*)?data\\/.*y(a)?ml$"
      ],
      "matchStringsStrategy": "combination",
      "matchStrings": [
        "-*?_image:\\s*(?<depName>.*)\\s",
        "-*?_version:\\s*(?<currentValue>.*)\\s"
      ],
      "datasourceTemplate": "docker"
    },
    {
      "matchStrings": [
        "\\s*image:\\s*(?<depName>.*):(?<currentValue>.*)\\s"
      ],
      "fileMatch": [
        "(.*)?data\\/.*y(a)?ml$"
      ],
      "datasourceTemplate": "docker"
    },
    {
      "fileMatch": [
        "(^|/|\\.)Dockerfile$",
        "(^|/)Dockerfile\\.[^/]*$"
      ],
      "matchStrings": [
        "# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s"
      ],
      "versioningTemplate": "{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"
    }'
{"fileMatch":["(.*)?data\\/.*y(a)?ml$"],"matchStringsStrategy":"combination","matchStrings":["-*?_image:\\s*(?<depName>.*)\\s","-*?_version:\\s*(?<currentValue>.*)\\s"],"datasourceTemplate":"docker"}
{"matchStrings":["\\s*image:\\s*(?<depName>.*):(?<currentValue>.*)\\s"],"fileMatch":["(.*)?data\\/.*y(a)?ml$"],"datasourceTemplate":"docker"}
{"fileMatch":["(^|/|\\.)Dockerfile$","(^|/)Dockerfile\\.[^/]*$"],"matchStrings":["# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s"],"versioningTemplate":"{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"}
```

```bash
RENOVATE_REGEX_MANAGERS: '[{"fileMatch":["(.*)?data\\/.*y(a)?ml$"],"matchStringsStrategy":"combination","matchStrings":["-*?_image:\\s*(?<depName>.*)\\s","-*?_version:\\s*(?<currentValue>.*)\\s"],"datasourceTemplate":"docker"}{"matchStrings":["\\s*image:\\s*(?<depName>.*):(?<currentValue>.*)\\s"],"fileMatch":["(.*)?data\\/.*y(a)?ml$"],"datasourceTemplate":"docker"}{"fileMatch":["(^|/|\\.)Dockerfile$","(^|/)Dockerfile\\.[^/]*$"],"matchStrings":["# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s"],"versioningTemplate":"{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"}]'
```

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Could not parse environment variable
       "val": "[{\"fileMatch\":[\"(.*)?data\\\\/.*y(a)?ml$\"],\"matchStringsStrategy\":\"combination\",\"matchStrings\":[\"-*?_image:\\\\s*(?<depName>.*)\\\\s\",\"-*?_version:\\\\s*(?<currentValue>.*)\\\\s\"],\"datasourceTemplate\":\"docker\"}{\"matchStrings\":[\"\\\\s*image:\\\\s*(?<depName>.*):(?<currentValue>.*)\\\\s\"],\"fileMatch\":[\"(.*)?data\\\\/.*y(a)?ml$\"],\"datasourceTemplate\":\"docker\"}{\"fileMatch\":[\"(^|/|\\\\.)Dockerfile$\",\"(^|/)Dockerfile\\\\.[^/]*$\"],\"matchStrings\":[\"# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\\\sENV .*?_VERSION=(?<currentValue>.*)\\\\s\"],\"versioningTemplate\":\"{{#if versioning}}{{versioning}}{{else}}semver{{/if}}\"}]",
       "envName": "RENOVATE_REGEX_MANAGERS"
```

</details>

**Comment by rarkins** [comment, marked as answer]
What you've copied into that RENOVATE_REGEX_MANAGERS does not appear to be valid JSON:

<img width="432" alt="image" src="https://user-images.githubusercontent.com/6311784/221913920-5821b63b-8c0c-478f-be7e-774905078938.png">

**Reply by rarkins** [reply]
`export RENOVATE_REGEX_MANAGERS='[{"fileMatch":["(.*)?data\\/.*y(a)?ml$"],"matchStringsStrategy":"combination","matchStrings":["-*?_image:\\s*(?<depName>.*)\\s","-*?_version:\\s*(?<currentValue>.*)\\s"],"datasourceTemplate":"docker"},{"matchStrings":["\\s*image:\\s*(?<depName>.*):(?<currentValue>.*)\\s"],"fileMatch":["(.*)?data\\/.*y(a)?ml$"],"datasourceTemplate":"docker"},{"fileMatch":["(^|/|\\.)Dockerfile$","(^|/)Dockerfile\\.[^/]*$"],"matchStrings":["# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s"],"versioningTemplate":"{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"}]'` works (added commas)

**Reply by Admirari80** [reply]
You are the best. Thank you.
